### ISSUE TITLE
Transparrent 3D Objects does not blend correctly in GLES2 (works in GLES3)

### ISSUE THREAD

**Issue by jesperkondrup** [godotengine/godot#29219]
3.1.1 Stable, windows 10. GLES2

When 3d objects have transparency in GLES2 they are not blending correctly - or at least it looks more correct in GLES3, where the blended effect is more additive. 

In this example I have 2 objects in front of each other with default spatial materials applied.  Transparency is on and I tried differrent combinations of Mix/alpha blend and depth draw modes but without any luck.

![image](https://user-images.githubusercontent.com/24936603/58419252-5f0c6200-808a-11e9-9246-3ff4c601cd08.png)

**Comment by akien-mga** [comment]
CC @clayjohn

**Comment by clayjohn** [comment]
I have a feeling this was fixed by https://github.com/godotengine/godot/pull/28520

Can you upload an MRP or try this out on a nightly build from https://hugo.pro/projects/godot-builds/

**Comment by jesperkondrup** [comment]
> I have a feeling this was fixed by #28520
> 
> Can you upload an MRP or try this out on a nightly build from https://hugo.pro/projects/godot-builds/

Yes it looks like it could be the same issue.
I have attached an MRP and provided a new image reference.

![image](https://user-images.githubusercontent.com/24936603/58459926-d2ff4680-812c-11e9-9456-746e0aadf4c5.png)

[GodotAdditiveTest_GLES2.zip](https://github.com/godotengine/godot/files/3226140/GodotAdditiveTest_GLES2.zip)

**Comment by jesperkondrup** [comment]
The issue is especially present if you have two materials with transparency in front of each other.

**Comment by akien-mga** [comment]
Thanks for the MRP, I can confirm that it's fixed in the current master branch (254286a).

For the reference, with GLES3 the scene raises a couple errors, but it's unrelated to this issue:
```
ERROR: mesh_surface_get_index_array: Condition ' surface->index_array_len == 0 ' is true. returned: PoolVector<uint8_t>()
   At: drivers/gles3/rasterizer_storage_gles3.cpp:3775.
ERROR: mesh_surface_get_index_array: Condition ' surface->index_array_len == 0 ' is true. returned: PoolVector<uint8_t>()
   At: drivers/gles3/rasterizer_storage_gles3.cpp:3775.
```

**Comment by akien-mga** [comment]
Fixed by #28520.
